test/onnx/CMakeLists.txt

```diff
@@ -22,15 +22,13 @@
 # THE SOFTWARE.
 #####################################################################################
 
-
 function(add_onnx_test TEST_NAME)
     rocm_add_test_executable(${TEST_NAME} ${ARGN})
     rocm_clang_tidy_check(${TEST_NAME})
     target_link_libraries(${TEST_NAME} migraphx_onnx migraphx_ref onnx_files)
     target_include_directories(${TEST_NAME} PUBLIC ../include include)
 endfunction()
 
-include(Embed)
```

Comment:
It seems strange to remove it here since its not included by a parent directory.

Reply:
I moved the Embed to src/CMakeLists.txt so functions are visible here.

Reply:
src/CMakeLists.txt is not a parent of the tests.

Reply:
Macros and functions in CMake are globally visible. The place of definition does not matter, and each module can call them. I could keep the include(Embed) as is, but since it can be included once I move it to src/CMakeLists.txt.

Reply:
However, they need to be defined before they are used. That's important. The `src` is added first, followed by `tests`. If they are added otherwise, then CMake will fail.

Reply:
It should be able to added in any order. So if tests are added first before src it would fail. So it needs to be included in the tests or in the top-level cmake.

Reply:
Moved to main CMakeLists.txt

Reply:
Reverted the change. Locally, it is building fine (Windows and Linux), but CI started failing to compile. Do not have time right now to investigate that.

Reply:
I dont think moving it to main cmake would cause a failure.

Reply:
I thought so too, but please look at the results of that commit:
https://github.com/ROCm/AMDMIGraphX/actions/runs/14878080192/job/41779412717
Because of my other priorities, I cannot investigate that further.